plugins/search-react/src/components/SearchTracker/SearchTracker.tsx

```diff
@@ -23,14 +23,18 @@ import { useSearch } from '../../context';
  */
 export const TrackSearch = ({ children }: { children: React.ReactChild }) => {
   const analytics = useAnalytics();
-  const { term } = useSearch();
+  const { term, result } = useSearch();
+
+  const numberOfResults = result.value?.numberOfResults ?? undefined;
 
   useEffect(() => {
     if (term) {
       // Capture analytics search event with search term provided as value
-      analytics.captureEvent('search', term);
+      analytics.captureEvent('search', term, {
+        value: numberOfResults,
+      });
```

Comment:
This is gonna cause multiple events to be captured right? First when triggering the search, and then again when a result is received. Is that the case, and is that fine?

Reply:
Thanks - good spot. For us this is fine as we're debouncing the analytics requests in our custom plugin - but for other users we should try to keep consistency. 

It would be nice if we could take the search term from the search engine response - then it updates with the same timing as the numberOfResults. I think this would likely be a breaking change though as some users may have a custom search engine (not one of Luna, PG, Elastic).

I'm looking for a solution in the React component (SearchTracker)

Reply:
I've added a hasFinishedLoading condition that stops the analytics event from firing until the loading state has changed from false -> true -> false.

With this we only get 1 analytics event per search - which is when the search engine responds with results

Previously it was when the search term updated (which I believe was coming from client side state - I've not dug deeply into this but I'm pretty sure this is the case).